File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    
    map<int,int>m;
    
    for(int i=0;i<numbers.size()-1;i++){
        for(int j=i+1;j<numbers.size();j++){
            m[numbers[i]+numbers[j]]=1;
        }
    }
    
    for(auto a:m){
        answer.push_back(a.first);
    }
    
    
    return answer;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = {};
        
        Set<Integer> s = new HashSet<>();

        int num;
        
        for(int i=0;i<numbers.length-1;i++){
            for(int j=i+1;j<numbers.length;j++){
                num = numbers[i]+numbers[j];
                s.add(num);
            }
        }
        
        return s.stream()
            .mapToInt(Integer::intValue)
            .sorted()
            .toArray();
    }

    
}